src/int.rs

```diff
@@ -3033,6 +3033,123 @@ impl PartialOrd<Int> for usize {
     }
 }
 
+
+const MAX_LIMB: u64 = !0 >> (64 - Limb::BITS);
+
+// do a sign-magnitude comparison
+fn eq_64(x: &Int, mag: u64, neg: bool) -> bool {
+    let x_is_zero = *x == 0;
+    let mag_is_zero = mag == 0;
+    if mag_is_zero || x_is_zero {
+        return mag_is_zero && x_is_zero
+    }
```

Comment:
Can't you just check the signs here and return false if they don't match? Something like this:

```rust
let mag_sign = if mag == 0 { 0 } else if neg { -1 } else { 1 };
if mag_sign != x.sign() { return false; }
```

I don't think the code really does much extra, but I think it's clearer than having a "has_right_sign" variable floating around, since if `has_right_sign == false` then the rest of the code is irrelevant.